docker-compose.yml

```diff
@@ -2,16 +2,45 @@ version: '3.8'
 services:
   mysql:
     image: mysql:8.0
-    command: --default-authentication-plugin=mysql_native_password
+    restart: always
+    command:
+      --default-authentication-plugin=mysql_native_password
+      --bind-address=0.0.0.0
     cap_add:
       - SYS_NICE
     ports:
-      - '3306:3306'
+      - 3306:3306
     environment:
       - MYSQL_ROOT_PASSWORD=default_password
       - MYSQL_DATABASE=checkpoint
     volumes:
       - mysql:/var/lib/mysql
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+    healthcheck:
+      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost", "-uroot", "-ppass"]
+      interval: 5s
+      timeout: 5s
+      retries: 20
+    network_mode: bridge
+
+  node:
+    build: .
+    command: yarn dev
+    restart: always
+    ports:
+      - 3000:3000
+    depends_on:
+      mysql:
+        condition: service_healthy
+    environment:
+      - DATABASE_URL=mysql://root:default_password@mysql:3306/checkpoint
+    links :
```

Comment:
```suggestion
    links:
```

Reply:
since Docker 1.10, it's generally recommended to use user-defined networks instead of links. In the context of your docker-compose.yml, because all services are part of the same Docker Compose app, they are already on the same network and can reach each other by service name, so you typically don't need to use links.